homeassistant/components/launch_library/sensor.py

```diff
@@ -1,69 +1,83 @@
+        [
+            NextLaunchSensor(coordinator),
+        ]
+    )
+
+
+class LLBaseEntity(CoordinatorEntity, SensorEntity):
+    """Sensor base entity."""
+
+    def __init__(self, coordinator: DataUpdateCoordinator) -> None:
+        """Initialize a Launch Library entity."""
+        super().__init__(coordinator)
+
+    def get_next_launch(self) -> Launch | None:
+        """Return next launch."""
+        return next((launch for launch in self.coordinator.data), None)
+
+
+class NextLaunchSensor(LLBaseEntity):
+    """Representation of the next launch sensor."""
+
+    _attr_icon = "mdi:orbit"
+    _attr_name = "Next launch"
+    _attr_unique_id = "next_launch"
 
+    @property
+    def native_value(self) -> str | None:
+        """Return the state of the sensor."""
+        next_launch = self.get_next_launch()
+        return next_launch.name if next_launch else None
 
-async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
-    """Create the launch sensor."""
-    name = config[CONF_NAME]
-    session = async_get_clientsession(hass)
-    launches = PyLaunches(session)
-
-    async_add_entities([LaunchLibrarySensor(launches, name)], True)
-
-
-class LaunchLibrarySensor(SensorEntity):
-    """Representation of a launch_library Sensor."""
-
-    _attr_icon = "mdi:rocket"
-
-    def __init__(self, api: PyLaunches, name: str) -> None:
-        """Initialize the sensor."""
-        self.api = api
-        self._attr_name = name
-
-    async def async_update(self) -> None:
-        """Get the latest data."""
-        try:
-            launches = await self.api.upcoming_launches()
-        except PyLaunchesException as exception:
-            _LOGGER.error("Error getting data, %s", exception)
-            self._attr_available = False
-        else:
-            if next_launch := next((launch for launch in launches), None):
-                self._attr_available = True
-                self._attr_native_value = next_launch.name
-                self._attr_extra_state_attributes = {
-                    ATTR_LAUNCH_TIME: next_launch.net,
-                    ATTR_AGENCY: next_launch.launch_service_provider.name,
-                    ATTR_AGENCY_COUNTRY_CODE: next_launch.pad.location.country_code,
-                    ATTR_STREAM: next_launch.webcast_live,
-                    ATTR_ATTRIBUTION: ATTRIBUTION,
-                }
+    @property
+    def extra_state_attributes(self) -> dict[str, Any]:
+        """Return the attributes of the sensor."""
+        next_launch = self.get_next_launch()
+        return (
+            {
+                ATTR_LAUNCH_TIME: next_launch.net,
+                ATTR_AGENCY: next_launch.launch_service_provider.name,
+                ATTR_AGENCY_COUNTRY_CODE: next_launch.pad.location.country_code,
+                ATTR_STREAM: next_launch.webcast_live,
+                ATTR_ATTRIBUTION: ATTRIBUTION,
+            }
+            if next_launch
+            else {}
+        )
```

Comment:
```suggestion
        return {
                ATTR_LAUNCH_TIME: next_launch.net,
                ATTR_AGENCY: next_launch.launch_service_provider.name,
                ATTR_AGENCY_COUNTRY_CODE: next_launch.pad.location.country_code,
                ATTR_STREAM: next_launch.webcast_live,
                ATTR_ATTRIBUTION: ATTRIBUTION,
            }
```

`ATTR_ATTRIBUTION` should not really be in here, but it already was in the old code, so we let it be for now and change that in another PR

Reply:
I will change it in the next PR

Reply:
Just move the attribute to a class attribute; for example:

```py
class NextLaunchSensor(LLBaseEntity):
    """Representation of the next launch sensor."""

	_attr_attribution = ATTRIBUTION
    _attr_icon = "mdi:orbit"
```

And remove it from the dictionary here.